pkg/scalers/postgresql_scaler.go

```diff
@@ -83,66 +116,26 @@ func NewPostgreSQLScaler(ctx context.Context, config *scalersconfig.ScalerConfig
 }
 
 func parsePostgreSQLMetadata(logger logr.Logger, config *scalersconfig.ScalerConfig) (*postgreSQLMetadata, kedav1alpha1.AuthPodIdentity, error) {
-	meta := postgreSQLMetadata{}
-
+	meta := &postgreSQLMetadata{}
 	authPodIdentity := kedav1alpha1.AuthPodIdentity{}
-
-	if val, ok := config.TriggerMetadata["query"]; ok {
-		meta.query = val
-	} else {
-		return nil, authPodIdentity, fmt.Errorf("no query given")
-	}
-
-	if val, ok := config.TriggerMetadata["targetQueryValue"]; ok {
-		targetQueryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, authPodIdentity, fmt.Errorf("queryValue parsing error %w", err)
-		}
-		meta.targetQueryValue = targetQueryValue
-	} else {
-		if config.AsMetricSource {
-			meta.targetQueryValue = 0
-		} else {
-			return nil, authPodIdentity, fmt.Errorf("no targetQueryValue given")
-		}
+	meta.triggerIndex = config.TriggerIndex
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, authPodIdentity, fmt.Errorf("error parsing postgresql metadata: %w", err)
 	}
 
-	meta.activationTargetQueryValue = 0
-	if val, ok := config.TriggerMetadata["activationTargetQueryValue"]; ok {
-		activationTargetQueryValue, err := strconv.ParseFloat(val, 64)
-		if err != nil {
-			return nil, authPodIdentity, fmt.Errorf("activationTargetQueryValue parsing error %w", err)
-		}
-		meta.activationTargetQueryValue = activationTargetQueryValue
+	if !config.AsMetricSource && meta.TargetQueryValue == 0 {
```

Comment:
Many scalers are checking AsMetricSource, would it be possible to simplify it and move this logic to Validate()? @wozniakjan

Reply:
valid point, although I am currently lacking good ideas on how to do that elegantly so I'm ok with either. Some amount of duplicate code in the scalers to check `AsMetricSource` or having generic way somewhere in the `TypedConfig` are both ok to me.

Reply:
What about back to this solution(https://github.com/kedacore/keda/pull/5819), which can make it more readable if moving config checking in Validate()?